### ISSUE TITLE
Visual representation of editing activity -- An LS graph?

### ISSUE THREAD

**Issue by dchandler** [ether/pad#103]
I mentioned this in an issue about "statistics", but feel that it merits its own issue.

The most effective way to describe a writing session really is visual. Lindgren and Sullivan (2002, and attached) describes the LS graph (Figure 6) -- http://dl.dropbox.com/u/5021551/Lindgren%20and%20Sullivan%202002.pdf. 

The LS graph displays a lot of information at once -- total characters typed over time, total characters that remain, where a user is editing, and when a "revision" occurs. Pedagogically speaking, there are plenty of writing researchers who have observed that showing students graphs like this can greatly improve how they write.

For example, this curve can show students their "writing strategy" -- are they planners? do they make edits as they go and obtain a perfect 1st draft? is their first draft "sloppy" and do they like to edit at the end? Etc.

In short, it's a way to have a conversation about how a student writes. The LS graph described in the paper is only for one user, but could be extended with multiple people quite easily.

In other threads, people mention how it would be nice to look at revisions. If somehow people could click on a point in the graph and then see a revision, this would provide a fantastic way to track editing activity of a pad throughout the writing session.

For the interested reader, Computer Keystroke Logging and Writing by Kirk Sullivan and Eva Lindgren describes the LS graph (which was the culmination of decades of writing research) and other methods for describing edits.

**Comment by mikkorantalainen** [comment]
Summary:
a graph with
    - x-axis: time from the creation of the document to
      the current time
    - y-axis: number of characters
    - three plots:
    -- number of edits from the start of the document (sum of
       insertions and deletions in characters, drawn as a thin line)
    -- length of the document in any given time (insertions minus
       deletions, drawn as a thick line)
    -- current position in characters from the start of the document
       (caret position, drawn as a thin line)
    - draw a thin vertical line for each saved version?

As a end result, the thick line will be always between two thin lines.

An obvious improvement would be to render the caret location for all editors in the same graph using the same color that's used for the actual text, too.

It would be great to replace the current "seek bar" which a graph like this (with a possibility to maximize the graph to get a more detailed picture).

**Comment by dchandler** [comment]
Mikkorantalainen -- You've absolutely read my mind both on this post and the other one regarding Diffs (http://github.com/ether/pad/issues/#issue/103). Thanks for helping translate what I wrote to specs.

I'm not one hundred percent sure what the thin vertical lines, which represent revisions, should be. What constitutes an interesting "revision" depends a lot on context. For some researchers, they've used "correction of spelling mistakes" or "reordering of sentences", but tracking these activities would be harder than tracking the basic three plots.

I completely agree that saves should also be represented. I think a simple star icon like in the time slider would suffice.
